onnxruntime/core/graph/graph_utils.h

```diff
@@ -86,20 +81,63 @@ bool GetRepeatedNodeAttributeValues(const Node& node,
   return false;
 }
 
-/** Removes the given Node from the Graph and keeps Graph consistent by rebuilding needed connections.
-    We support the removal of the Node as long as the following conditions hold:
-    - There should be no implicit inputs.
-    - Only one of the outputs is used by downstream operators (but it can have multiple output edges).
-    - If the Node has a single incoming node (and possibly multiple initializers), we can remove the Node and
-      connect its incoming node to its outgoing nodes.
-    - If the Node has a single initializer as input, we remove the Node and feed the initializer as input to its
-      output nodes. */
-bool RemoveNode(Graph& graph, Node& node);
+/** Tests if we can remove a node and merge its input edge (if any) with its output edges.
+Conditions:
+ - Only one of the node's outputs is used by downstream operators
+   - multiple edges for the single used output are allowed
+ - The node must have zero or one input edges 
+   - we merge any input edge with the output edges for the single output that is used
+ - If the node has no input edges it must have a single input
+   - this will be a graph input or initializer
+ - The node must not produce a graph output
+   - the node removal will result in that output name not being produced
+ - Removing the node won't break a subgraph that consumes the node's output
+*/
+bool CanRemoveNodeAndMergeEdges(const Graph& graph, const Node& node);
+
+/** Removes the given Node from the Graph and merges its input edge (if present) with the edges for its output.
+See CanRemoveNodeAndMergeEdges for the conditions that must be satisfied in order to remove the node.*/
+bool RemoveNodeAndMergeEdges(Graph& graph, Node& node);
```

Comment:
The name and comment is not very descriptive for the case of no input edge.

Reply:
Changed to call 'RemoveNode' with a lot more documentation.